### ISSUE TITLE
lxc-attach: Failed to get cgroup for controller all

### ISSUE THREAD

**Issue by asakura** [lxc/lxc#418]
```sh-session
$ sudo lxc-create -t ubuntu -n test-1
....
sudo lxc-start -n test-1
sudo lxc-attach -n test-1 -- id
lxc_container: cgmanager.c: cgm_attach: 1291 Failed to get cgroup for controller all
lxc_container: attach.c: lxc_attach: 847 error communicating with child process
```

Software versions:
* `Linux wayar 3.16.0-29-generic #39-Ubuntu SMP Mon Dec 15 22:27:29 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux`
* `LXC version is 1.1.0~alpha2-0ubuntu3`
* `CGManager version is 0.35-1~ubuntu14.10.1~ppa1`

Is there any other workaround for that instead of `dpkg -i cgmanager_0.32-4ubuntu1.2_amd64.deb libcgmanager0_0.32-4ubuntu1.2_amd64.deb`?

**Comment by stgraber** [comment]
I assume you tried rebooting the system already?

Can you paste "sudo cat /proc/self/cgroup" and "sudo cat /proc/$(pidof cgmanager/cgroup"?

**Comment by asakura** [comment]
Yes, I rebooted box.

```sh-session
$ sudo cat /proc/self/cgroup
12:name=systemd:/user.slice/user-1000.slice/session-c2.scope
11:perf_event:/user.slice/user-1000.slice/session-c2.scope
10:net_prio:/user.slice/user-1000.slice/session-c2.scope
9:net_cls:/user.slice/user-1000.slice/session-c2.scope
8:memory:/user.slice/user-1000.slice/session-c2.scope
7:hugetlb:/user.slice/user-1000.slice/session-c2.scope
6:freezer:/user.slice/user-1000.slice/session-c2.scope
5:devices:/user.slice/user-1000.slice/session-c2.scope
4:cpuacct:/user.slice/user-1000.slice/session-c2.scope
3:blkio:/user.slice/user-1000.slice/session-c2.scope
2:cpu:/user.slice/user-1000.slice/session-c2.scope
1:cpuset:/user.slice/user-1000.slice/session-c2.scope
$ sudo cat /proc/$(pidof cgmanager)/cgroup
12:name=systemd:/
11:perf_event:/
10:net_prio:/
9:net_cls:/
8:memory:/
7:hugetlb:/
6:freezer:/
5:devices:/
4:cpuacct:/
3:blkio:/
2:cpu:/
1:cpuset:/
```

**Comment by stgraber** [comment]
Hmm, that all looks good... well, except for you running with the latest cgmanager on stable Ubuntu (which didn't include it) and using the default LXC from that Ubuntu release, but that should still work.

Can you paste the content of /var/log/upstart/cgmanager.log?

**Comment by asakura** [comment]
```sh-session
$ cat /var/log/upstart/cgmanager.log
cgmanager:list_keys_main: Could not determine the requested cgroup (all:)
cgmanager: Found no cgroup entry for pid 10348 controller all

cgmanager:get_pid_cgroup_abs_main: Could not determine the requestor's cgroup for all
cgmanager: Found no cgroup entry for pid 13191 controller all

cgmanager:get_pid_cgroup_abs_main: Could not determine the requestor's cgroup for all
cgmanager: Found no cgroup entry for pid 424 controller all

cgmanager:get_pid_cgroup_abs_main: Could not determine the requestor's cgroup for all
```

**Comment by asakura** [comment]
Yeah, right - cgmanager comes from `ppa:ubuntu-lxc/stable` and lxc comes from Ubuntu's upstream (https://launchpad.net/ubuntu/+source/lxc/1.1.0~alpha2-0ubuntu3.1). I dropped that PPA's record and made cgmanager package's downgrade. Now everything work perfect. Thank you.

**Comment by stgraber** [comment]
An alternative if you want the recent cgmanager would be to use ppa:ubuntu-lxc/daily which has it too and a recent LXC which shouldn't have whatever bug you ran into (since I'm running that and lxc-attach sure works here).

**Comment by asakura** [comment]
I'll try that. Thanks.

**Comment by hallyn** [comment]
This should be fixed by upstream commit 5d9dd3d8f72cbcf67c5d375a83a4ed4372932a11, which apparently has not been backported to utopic.
